docs/strategies/circuit-breaker.md

```diff
@@ -99,3 +99,229 @@ await manualControl.CloseAsync();
+
+### 2 - Using different duration for breaks
+
+In case of Retry you can specify dynamically the sleep duration via the `DelayGenerator`.
+In case of Circuit Breaker the `BreakDuration` is considered constant (can't be changed between breaks).
+
+❌ DON'T
+
+Use `Task.Delay` inside `OnOpened`
+
+<!-- snippet: circuit-breaker-anti-pattern-2 -->
+```cs
+static IEnumerable<TimeSpan> GetSleepDuration()
+{
+    for (int i = 1; i < 10; i++)
+    {
+        yield return TimeSpan.FromSeconds(i);
+    }
+}
+
+var sleepDurationProvider = GetSleepDuration().GetEnumerator();
+sleepDurationProvider.MoveNext();
+
+var circuitBreaker = new ResiliencePipelineBuilder()
+    .AddCircuitBreaker(new()
+    {
+        ShouldHandle = new PredicateBuilder().Handle<HttpRequestException>(),
+        BreakDuration = TimeSpan.FromSeconds(0.5),
+        OnOpened = async args =>
+        {
+            await Task.Delay(sleepDurationProvider.Current);
+            sleepDurationProvider.MoveNext();
+        }
+
+    })
+    .Build();
+```
+<!-- endSnippet -->
+
+**Reasoning**:
+- The lowest value of break duration is half second
+  - That means each sleep is actually takes `sleepDurationProvider.Current` + half second
+- You might think that setting the `BreakDuration` to the `sleepDurationProvider.Current` solves the problem
+  - But it doesn't since the `BreakDuration` is set only once and not re-evalutated at every break
+
+<!-- snippet: circuit-breaker-anti-pattern-2-ext -->
+```cs
+circuitBreaker = new ResiliencePipelineBuilder()
+    .AddCircuitBreaker(new()
+    {
+        ShouldHandle = new PredicateBuilder().Handle<HttpRequestException>(),
+        BreakDuration = sleepDurationProvider.Current,
+        OnOpened = async args =>
+        {
+            Console.WriteLine($"Break: {sleepDurationProvider.Current}");
+            sleepDurationProvider.MoveNext();
+        }
+
+    })
+    .Build();
+```
+<!-- endSnippet -->
+
+✅ DO
+
+The `CircuitBreakerStartegyOptions` currently do not provide a way to define break durations dynamically.
+This might be re-evaluted later until that use the first example as a workaround with caution.
+
+### 3 - Wrapping each endpoint with a circuit breaker
+
+Imagine that you have to call N number of services via `HttpClient`s.
+You want to decorate all downstream calls with the service-aware Circuit Breaker.
+
+❌ DON'T
+
+Use a collection of Circuit Breakers and explicitly call `ExecuteAsync`
+
+<!-- snippet: circuit-breaker-anti-pattern-3 -->
+```cs
+// Defined in a common place
+var uriToCbMappings = new Dictionary<Uri, ResiliencePipeline>
+{
+    { new Uri("https://downstream1.com"), GetCircuitBreaker() },
+    // ...
+    { new Uri("https://downstreamN.com"), GetCircuitBreaker() }
+};
+
+// Used in the downstream 1 client
+var downstream1Uri = new Uri("https://downstream1.com");
+await uriToCbMappings[downstream1Uri].ExecuteAsync(CallXYZOnDownstream1, CancellationToken.None);
+```
+<!-- endSnippet -->
+
+**Reasoning**:
```

Comment:
## Reasoning

- Whenever you use an `HttpClient`, you must have a reference to the `uriToCbMappings`.
- It's your responsibility to decorate each network call with the corresponding circuit breaker.